### DISCUSSION TITLE
How to configure different storage class for each instance/replica

### DISCUSSION THREAD

**Discussion by xyzmurali** [cloudnative-pg/cloudnative-pg#3317] [Q&A]
Hi,
Is it possible to configure different Storage classes for each instance/replica of the Postgres cluster?

The documentation mentions using a separate storage class for WAL Storage using **walStorage** option. But how can we specify different storage classes for different instances/replicas within the same Postgres cluster?

Our use case: We want to create a cluster with 3 instances/replicas and are looking at the option of linking each of these instances to PVs provisioned by 3 different storage classes from 3 different storages to support HA. 

Do you know of any other method to achieve this?

**Comment by gbartolini** [comment]
A High Availability cluster in our vision means that any instance can become at any time the primary, without impacting on performance. Each instance must be identical and we do not care which one is primary.

For this reason, they all have the same storage classes per volume type (PGDATA, WALs and shortly tablespaces too).

You can achieve something similar with [replica clusters](https://cloudnative-pg.io/documentation/current/replica_cluster/) but ... they are not for automated failover supported HA.

Why do you need 3 different storage classes, if you don't mind? I would like to understand.

**Reply by Ofirfr** [reply]
Hi, I was just having the same Idea as the Author.
In my compnay we have 2 storage machines which are separate, and when we deploy our DBs (currently with Patroni on VMs) we make sure that at least one replica is on a different storage machine in case of some failure it might have (and it usually saves us from major downtime).
We would like to implement the same type of HA with CNPG, where we would like to have replicas on different storage machines, which can be done by using different storage classes for each replica.
Is it considered in the Roadmap or anything? Or, if you have any other ideas for me, I would like to hear them!

**Comment by xyzmurali** [comment]
Hi, Thanks for your response. I was thinking in terms of High Availability in case if we have only one cluster with storage class created from network storage or similar. If that one storage for some reason goes offline or bad the entire pg cluster goes bad. But if we have the instances pointing to different storage classes coming from different devices, then we have another level of HA.
